openedx/core/djangoapps/content/learning_sequences/api/processors/special_exams.py

```diff
@@ -54,26 +57,49 @@ def exam_data(self, pruned_course_outline: UserCourseOutlineData) -> SpecialExam
                         continue
 
                     special_exam_attempt_context = None
-                    try:
-                        # Calls into edx_proctoring subsystem to get relevant special exam information.
-                        # This will return None, if (user, course_id, content_id) is not applicable.
-                        special_exam_attempt_context = get_attempt_status_summary(
-                            self.user.id,
-                            str(self.course_key),
-                            str(sequence.usage_key)
-                        )
-                    except ProctoredExamNotFoundException:
-                        log.info(
-                            'No exam found for {sequence_key} in {course_key}'.format(
-                                sequence_key=sequence.usage_key,
-                                course_key=self.course_key
+
+                    # if exam waffle flag enabled, then use exams logic
+                    if exams_ida_enabled(self.course_key):
+                        # add short description based on exam type
+                        if sequence.exam.is_practice_exam:
```

Comment:
This is just user facing text on the outline. I'd recommend we do this is the learning-mfe and not the backend so the translations are a bit closer too where they are actually rendered.

Reply:
Ah okay! So just so I understand, would all of this need to happen in the learning-mfe? Or just the content, while the flag logic remains here? Thanks for clarifying, @Zacharis278.

Reply:
well my gut reaction is this should but UI code but hold that thought for a min it's possible I'm wrong in how this endpoint is intended to be used☝️

Reply:
ok chased this thread for a bit, exams aren't the only thing using this description field, its default is actually the UI string for the grading policy ([CourseBlockSerializer](https://github.com/openedx/edx-platform/blob/379b9e4f7d959a61c79a79b29e7ed1081c50a6fc/lms/djangoapps/course_home_api/outline/serializers.py#L13)). So, adding fields to independently pass exam type up to the frontend does seem to make things unnecessarily complicated. I'm happy to just keep this on the backend for simplicity.

We do want these strings to get translated tho, is something translating the noop later down the line? I think these can just be a normal gettext() call.

Reply:
Thank you for looking into this and clarifying! And yes, it is being translated later so this is probably unnecessary here. Thanks again!